### ISSUE TITLE
ErrorHandler - Failed Recovery Still Commits Kafka Events

### ISSUE THREAD

**Issue by PatrickTheFirst** [spring-projects/spring-kafka#1266]
**Problem to be Solved**
My team and me created some kind of error recovery for Kafka messages in case our application is not able to consume for various reasons.

Basically our solution stores erroneous Kafka messages in an error table that we can process later. As a result those erroneous Kafka messages can be committed and life can continue.

**Bug**
When our `DeadletterPersistenceRecoverer` fails for some reason with an exception it is crucial that no commit happens because otherwise we lose messages.

That worked fine with version **2.2.8** and does no longer work since version **2.2.9** and the following fix: #1209 

Now all exceptions of the recoverer are caught and the `SeekToCurrentErrorHandler` assumes recovery was ok and commits.

**Expected**
Failed recoveries should never ever lead to committed Kafka events.

**Here is some code:**
```
@Bean
    public ErrorHandler recoveryErrorHandler(DeadLetterPersistenceRecoverer deadLetterPersistenceRecoverer,
                                             KafkaErrorPersistenceConfigurationProperties persistenceProperties) {

        SeekToCurrentErrorHandler errorHandler = new SeekToCurrentErrorHandler(deadLetterPersistenceRecoverer,
            persistenceProperties.getRetries() + 1);
        errorHandler.setCommitRecovered(true);

        return errorHandler;
    }
```
and the recoverer...
```
public class DeadLetterPersistenceRecoverer implements BiConsumer<ConsumerRecord<?, ?>, Exception> {

    private static final Logger LOGGER = LoggerFactory.getLogger(DeadLetterPersistenceRecoverer.class);

    private final KafkaErrorRepository kafkaErrorRepository;

    public DeadLetterPersistenceRecoverer(KafkaErrorRepository kafkaErrorRepository) {
        this.kafkaErrorRepository = kafkaErrorRepository;
    }

    @Override
    @Transactional
    public void accept(ConsumerRecord<?, ?> consumerRecord, Exception e) {

        if (LOGGER.isInfoEnabled()) {
            LOGGER.info("Failed to consume kafka message with key [{}] of topic [{}]. Writing message to error table...",
                consumerRecord.key(), consumerRecord.topic());
        }

        KafkaError kafkaError = getKafkaErrorByConsumerRecord(consumerRecord, e);
        kafkaErrorRepository.save(kafkaError);

        LOGGER.info("Successfully stored message with key [{}] in error table.", consumerRecord.key());
    }
}
```

**Comment by garyrussell** [comment]
Duplicate of https://github.com/spring-projects/spring-kafka/issues/1259

This is already fixed; please try 2.2.10.BUILD-SNAPSHOT; we will release 2.2.10 next week.

**Comment by garyrussell** [comment]
2.2.10 is on its way to maven central and should be there shortly.
